### PR TITLE
[opensprinkler] Add water level channel

### PR DESCRIPTION
**Pull request by pail23**
This PR adds the water level channel to the opensprinkler device. This channel is calculated based on weather data and influences how long the watering will last

### CODE REVIEW

File: bundles/org.openhab.binding.opensprinkler/README.md [hunk 1]
```diff
@@ -66,6 +66,8 @@ The following is supported by the `device` thing, but only when connected using
 | rainsensor      | Switch                 | RO | This channel indicates whether rain is detected by the device or not.     |
 | currentDraw     | Number:ElectricCurrent | RO | Shows the current draw of the device. If the device does not have sensors |
 |                 |                        |    | for this metric, the channel will not be available.                       |
+| waterlevel      | Number                 | RO | This channel shows the current water level. The water level calculated    |
```
Comment (fwolter) [inline bundles/org.openhab.binding.opensprinkler/README.md:69]: Can you mention that it is percent?
```suggestion
| waterlevel      | Number:Dimensionless                 | RO | This channel shows the current water level. The water level calculated    |
```

**Comment by FlorianSW** [comment]
Big sorry for not coming back to this earlier :( I'll look into this right now :)

File: bundles/org.openhab.binding.opensprinkler/README.md [hunk 2]
```diff
@@ -66,6 +66,8 @@ The following is supported by the `device` thing, but only when connected using
 | rainsensor      | Switch                 | RO | This channel indicates whether rain is detected by the device or not.     |
 | currentDraw     | Number:ElectricCurrent | RO | Shows the current draw of the device. If the device does not have sensors |
 |                 |                        |    | for this metric, the channel will not be available.                       |
+| waterlevel      | Number:Dimensionless                 | RO | This channel shows the current water level. The water level calculated    |
```
Comment (FlorianSW) [inline bundles/org.openhab.binding.opensprinkler/README.md:69]: probably align with other cells :)

File: bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/api/OpenSprinklerHttpApiV100.java [hunk 3]
```diff
@@ -208,17 +215,10 @@ public int getNumberOfStations() throws CommunicationApiException {
...
-        try {
-            returnContent = http.sendHttpGet(getBaseUrl() + CMD_OPTIONS_INFO, null);
-        } catch (Exception exp) {
-            throw new CommunicationApiException(
-                    "There was a problem in the HTTP communication with the OpenSprinkler API: " + exp.getMessage());
-        }
 
         try {
-            this.firmwareVersion = Parse.jsonInt(returnContent, JSON_OPTION_FIRMWARE_VERSION);
+            JoResponse info = getOptions();
```
Comment (FlorianSW) [inline bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/api/OpenSprinklerHttpApiV100.java:220]: awesome for refactoring this as well :)

File: bundles/org.openhab.binding.opensprinkler/README.md (hunk 2 above)
Comment (pail23) [reply bundles/org.openhab.binding.opensprinkler/README.md]: Fixed.

File: bundles/org.openhab.binding.opensprinkler/README.md [hunk 4]
```diff
@@ -66,6 +66,8 @@ The following is supported by the `device` thing, but only when connected using
 | rainsensor      | Switch                 | RO | This channel indicates whether rain is detected by the device or not.     |
 | currentDraw     | Number:ElectricCurrent | RO | Shows the current draw of the device. If the device does not have sensors |
 |                 |                        |    | for this metric, the channel will not be available.                       |
+| waterlevel      | Number:Dimensionless   | RO | This channel shows the current water level. The water level calculated    |
+|                 |                        |    | based on the weather and influences the duration of the water programs.   |
```
Comment (fwolter) [inline bundles/org.openhab.binding.opensprinkler/README.md:70]: Can you mention, that this is percent?

File: bundles/org.openhab.binding.opensprinkler/src/main/resources/ESH-INF/thing/thing-types.xml [hunk 5]
```diff
@@ -85,6 +86,14 @@
 		<state readOnly="true"/>
 	</channel-type>
 
+
+	<channel-type id="waterlevel">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Water level</label>
+		<description>The current water level</description>
```
Comment (fwolter) [inline bundles/org.openhab.binding.opensprinkler/src/main/resources/ESH-INF/thing/thing-types.xml:93]:
```suggestion
		<description>The current water level in percent</description>
```

File: bundles/org.openhab.binding.opensprinkler/src/main/resources/ESH-INF/thing/thing-types.xml [hunk 6]
```diff
@@ -85,6 +86,14 @@
 		<state readOnly="true"/>
 	</channel-type>
 
+
+	<channel-type id="waterlevel">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Water level</label>
```
Comment (fwolter) [inline bundles/org.openhab.binding.opensprinkler/src/main/resources/ESH-INF/thing/thing-types.xml:92]: Words in labels should be capitalized (except prepositions and so on). See https://www.openhab.org/docs/developer/bindings/thing-xml.html#formatting-labels-and-descriptions

**Review by fwolter — CHANGES_REQUESTED** [review]
There are some formatting issues. You can view them with `mvn spotless:check -Dspotless.check.skip=false` and fix them with `mvn spotless:apply`.

File: bundles/org.openhab.binding.opensprinkler/src/main/resources/ESH-INF/thing/thing-types.xml (hunk 5 above)
Comment (pail23) [reply bundles/org.openhab.binding.opensprinkler/src/main/resources/ESH-INF/thing/thing-types.xml]: done

File: bundles/org.openhab.binding.opensprinkler/README.md (hunk 4 above)
Comment (pail23) [reply bundles/org.openhab.binding.opensprinkler/README.md]: yes, done

**Comment by pail23** [comment]
Thanks @fwolter for the review. The findings should be fixed now.

**Review by fwolter — APPROVED** [review]

**Pull request merged**